### PR TITLE
Fix bug where finding the max heatmap location after non-maximum suppression returned empty list

### PR DESCRIPTION
**Pull request by robertklee**
`np.where(peaks == peaks.max())` was returning an empty list for some reason. I suspect it is due to floating point equality comparison.

This PR uses a more robust way that guarantees that a result will be returned.

I've also looked into partially patching `PCK`.

### CODE REVIEW

**Review by nicolepeverley — APPROVED** [review]
LGTM 👍  the only other change I had was using the head as a secondary measure but we could also just roll with this for now

**Comment by robertklee** [comment]
@nicolepeverley i like your changes better. Let's merge both and keep yours for PCK-related fixes for now? I know Corey was also taking a look at it from the original paper.

File: evaluation.py [hunk 1]
```diff
@@ -229,18 +229,16 @@ def heatmaps_to_keypoints(self, heatmaps, threshold=HM_TO_KP_THRESHOLD):
...
             # Choose the max point in heatmap (we only pick 1 keypoint in each heatmap)
             # and get its coordinates and confidence
-            y, x = np.where(peaks == peaks.max())
+            y, x = np.unravel_index(np.argmax(peaks), peaks.shape)
 
-            if int(x[0]) > 0 and int(y[0]) > 0:
-                x_new = int(x[0])
-                y_new = int(y[0])
-                conf_new = peaks[y_new, x_new]
+            if peaks[y, x] > HM_TO_KP_THRESHOLD / 5:
```
Comment (julianrocha) [inline evaluation.py]: Why divide this by 5? Can we just change the constant?

**Review by julianrocha — APPROVED** [review]
Nice! So for the PCK stuff we will roll with Nicoles PR? I left one comment but looks good otherwise

File: evaluation.py (hunk 1 above)
Comment (robertklee) [reply evaluation.py]: I wasn't sure if non-maximum suppression would reduce the max intensity so i picked an arbitrary number. anything below that threshold should have already been reset to 0 though

**Comment by robertklee** [comment]
yeah, the pck still is not fixed with this PR. See attached.

[pck.csv](https://github.com/robertklee/COCO-Human-Pose/files/6292501/pck.csv)

**Pull request merged**